### PR TITLE
Docs: Deprecate --default-language and set version for --icu-language

### PR DESCRIPTION
**Pull request by Simran-B**
Backport of #16039

Follow-up PR to https://github.com/arangodb/arangodb/pull/15912

Docs PR: https://github.com/arangodb/docs/pull/946

### CODE REVIEW

**Comment by Simran-B** [comment]
`--default-language` will formally be deprecated in v3.10.0, rendering this PR obsolete.

**Comment by Simran-B** [comment]
Actually, the "introduced in 3.9.1" is still needed, re-opening.

**Review by KVS85 — APPROVED** [review]
LGTM.

File: lib/ApplicationFeatures/LanguageFeature.cpp [hunk 1]
```diff
@@ -126,9 +126,11 @@ void LanguageFeature::collectOptions(
       new StringParameter(&_defaultLanguage),
       arangodb::options::makeDefaultFlags(arangodb::options::Flags::Hidden));
```
Comment (alexbakharew) [inline lib/ApplicationFeatures/LanguageFeature.cpp:127]: Should we add here info about deprecation of this option in 3.10?

Comment (KVS85) [reply lib/ApplicationFeatures/LanguageFeature.cpp:127]: Not in 3.9 branch.

**Review by KVS85 — APPROVED** [review]
LGTM.

**Review by alexbakharew — APPROVED** [review]
LGTM

**Review by ansoboleva — APPROVED** [review]
reviewed

**Pull request merged**